backend/src/auth/data_model/UserModel.ts

```diff
@@ -0,0 +1,31 @@
+import { Email } from "../../../../domain_model/Email"
+import { Uid } from "../../../../domain_model/Uid"
+
+
+export class UserModel {
+    id: Uid;
+    email: Email;
+    hashedPassword: string;
+    salt: string;
+
+    constructor(data: Pick<UserModel, "id" | "email" | "hashedPassword" | "salt">) {
+        this.id = data.id;
+        this.email = data.email;
+        this.hashedPassword = data.hashedPassword;
```

Comment:
At what point will we generate the hashedPassword from the raw password? Could we take the raw password as a parameter and do the hashing here?

Reply:
We'd do that when a user first registers, or changes their password.  I didn't stick that in the constructor as I imagine those being explicit methods.